js/index.js

```diff
@@ -1,42 +1,72 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
+    console.log('Calculating subtotal, yey!');
 
-  //... your code goes here
+    //getting the first element with class="price" and "quantity" from the query.
+    const priceSpanElement = product.querySelector('.price span');
+    const quantityInputElement = product.querySelector('.quantity input')
+
+    //converting the priceSpanElement string into a number
+    const price = parseFloat(priceSpanElement.innerText);
+
+    //setting the quantity value as a number
+    const quantity = quantityInputElement.valueAsNumber;
+
+    //calculating the subtotal
+    const subTotal = price * quantity
+
+    //getting the first element with class="subtotal" from the query.
+    const subTotalElement = product.querySelector('.subtotal span');
+
+    //sending subtotalelement to inner text
+    subTotalElement.innerText = subTotal;
+    return subTotal;
 }
 
+
+//   // code in the following two lines is added just for testing purposes.
+//   // it runs when only iteration 1 is completed. at later point, it can be removed.
+
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+    const singleProduct = document.querySelector('.product');
+    updateSubtotal(singleProduct);
+
+    // ITERATION 2
+
+
+    const products = document.getElementsByClassName('product')
+    let totalValue = 0;
+
+    for (let counter = 0; counter < products.length; counter++) {
+        const element = products[counter];
+        totalValue += updateSubtotal(element);
+    }
+
+
 
-  // ITERATION 2
-  //... your code goes here
+    //     // ITERATION 3
+    document.querySelector('#total-value span').innerHTML = totalValue;
 
-  // ITERATION 3
-  //... your code goes here
 }
 
 // ITERATION 4
 
```

Comment:
Nice work!

I'll be adding a working solution of the bonuses below so you can go over it in your own time
```
// ITERATION 4

function removeProduct(event) {
  const target = event.currentTarget;
  const row = target.parentNode.parentNode;
  const parent = row.parentNode;
  parent.removeChild(row);

  calculateAll();
}

// ITERATION 5

function createProduct() {
  const createRow = document.querySelector('.create-product');
  let newProdNameInput = createRow.querySelector('input');
  let newProdNameValue = newProdNameInput.value;
  let newProdPriceInput = createRow.querySelector("input[type='number']");
  let newProdPriceValue = Number(newProdPriceInput.valueAsNumber).toFixed(2);

  const newTableRow = document.createElement('tr');
  newTableRow.className = 'product';
  newTableRow.innerHTML = `
    <td class="name">
      <span>${newProdNameValue}</span>
    </td>
    <td class="price">$<span>${newProdPriceValue}</span></td>
    <td class="quantity">
      <input type="number" value="0" min="0" placeholder="Quantity" />
    </td>
    <td class="subtotal">$<span>0</span></td>
    <td class="action">
      <button class="btn btn-remove">Remove</>
    </td>
  `;

  // get the parent of this newly created row
  const parent = document.querySelector('#cart tbody');

  // append the newly created row to the parent
  parent.appendChild(newTableRow);

  // make sure remove button inherits the same behavior as other remove buttons
  const removeBtn = newTableRow.querySelector('.btn-remove');
  removeBtn.addEventListener('click', removeProduct);

  // clean the fields
  newProdNameInput.value = '';
  newProdPriceInput.value = 0;
}

window.addEventListener('load', () => {
  const calculatePricesBtn = document.getElementById('calculate');
  calculatePricesBtn.addEventListener('click', calculateAll);

  const removeBtns = document.querySelectorAll('.btn-remove');
  for (let removeBtn of removeBtns) {
    removeBtn.addEventListener('click', removeProduct);
  }

  const createBtn = document.getElementById('create');
  if (createBtn) {
    createBtn.addEventListener('click', createProduct);
  }
});